ibis/backends/polars/__init__.py

```diff
@@ -395,9 +395,7 @@ def compile(
         return translate(node, ctx=ctx)
 
     def _get_schema_using_query(self, query: str) -> sch.Schema:
-        return schema_from_polars(
-            self._context.execute(f"SELECT * FROM ({query}) LIMIT 0").schema
-        )
+        return schema_from_polars(self._context.execute(query).schema)
```

Comment:
`execute` is lazy, so there's no need to worry about executing the entire query.